sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -316,168 +310,161 @@ static int register_device()
   MQTTClient_free(topic);
-  return rc;
+  return;
 }
 
-static az_result parse_operation_message(
+static void parse_operation_message(
     char* topic,
     int topic_len,
     MQTTClient_message const* message,
     az_iot_provisioning_client_register_response* response,
     az_iot_provisioning_client_operation_status* operation_status)
 {
   int rc;
-
-  if (topic_len == 0)
-  {
-    // The length of the topic if there are one or more NULL characters embedded in topic,
-    // otherwise topic_len is 0.
-    topic_len = (int)strlen(topic);
-  }
-
   az_span topic_span = az_span_init((uint8_t*)topic, topic_len);
   az_span message_span = az_span_init((uint8_t*)message->payload, message->payloadlen);
-  
-  print_az_span("Topic: ", topic_span);
 
-  // Parse the incoming message and payload
   if (az_failed(
           rc = az_iot_provisioning_client_parse_received_topic_and_payload(
               &provisioning_client, topic_span, message_span, response)))
   {
-    (void)printf("Message from unknown topic: az_result return code 0x%04x .\n", rc);
-    return rc;
+    LOG_ERROR("Message from unknown topic: az_result return code 0x%04x .\n", rc);
+    exit(rc);
   }
-  print_az_span("Received payload:\n", message_span);
-  (void)printf("Response status: %d\n", response->status);
+  LOG_SUCCESS("Received response:\n");
+  print_az_span("\t Topic: ", topic_span);
+  print_az_span("\t Payload: ", message_span);
+  (void)printf("\t Response status: %d\n", response->status);
 
-  // Parse the operation status from a string to an enum
   if (az_failed(rc = az_iot_provisioning_client_parse_operation_status(response, operation_status)))
   {
-    (void)printf("Failed to parse operation_status: az_result return code 0x%04x .\n", rc);
-    return rc;
+    LOG_ERROR("Failed to parse operation_status: az_result return code 0x%04x .\n", rc);
+    exit(rc);
   }
 
-  return AZ_OK;
+  return;
 }
 
-static int send_operation_query_message(
+static void send_operation_query_message(
     az_iot_provisioning_client_register_response const* response)
 {
   int rc;
 
-  (void)printf("Sending operation query message.\n");
-
-  // Get the topic to send the query message
   if (az_failed(
           rc = az_iot_provisioning_client_query_status_get_publish_topic(
-              &provisioning_client, response, query_topic, sizeof(query_topic), NULL)))
+              &provisioning_client, response, query_topic_buffer, sizeof(query_topic_buffer), NULL)))
   {
-    (void)printf("Unable to get query status publish topic: az_result return code 0x%04x .\n", rc);
-    return rc;
+    LOG_ERROR("Unable to get query status publish topic: az_result return code 0x%04x .\n", rc);
+    exit(rc);
   }
 
   // IMPORTANT: Wait the recommended retry-after number of seconds before query
-  (void)printf("Querying after %u seconds...\n", response->retry_after_seconds);
+  (void)printf("\t Querying after %u seconds...\n", response->retry_after_seconds);
```

Comment:
Should we maybe have `LOG_SUCCESS`, `LOG_ERROR` and `LOG_MESSAGE`? `LOG_MESSAGE` would align but remove the SUCCESS: prefix. Something about mixing log macro's and other `printf`'s doesn't sit well.

Reply:
haha, attempted to fix this in next push before seeing your comments.  please take a look.